src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/DataSetQueryService.cs

```diff
@@ -348,6 +322,46 @@ private static List<Sort> GetSorts(
         return sorts;
     }
 
+    private static bool TryGetSortColumn(
+        DataSetQuerySort sort,
+        ColumnMappings columnMappings,
+        [NotNullWhen(true)]
+        out string? column)
+    {
+        column = null;
+
+        if (ReservedSorts.TryGetValue(sort.Field, out var reservedColumn))
+        {
+            column = reservedColumn;
+            return true;
+        }
+
+        var fieldParts = sort.Field.Split('|', 2);
+
+        if (fieldParts.Length != 2)
+        {
+            return false;
+        }
+
+        var fieldType = fieldParts[0];
+        var fieldId = fieldParts[1];
+
+        switch (fieldType)
+        {
+            case SortTypeFilter when columnMappings.Filters.TryGetValue(fieldId, out var filterColumn):
```

Comment:
Actually here I think this can be reduced to:

```
        switch (fieldType)
        {
            case SortTypeFilter:
                return columnMappings.Filters.TryGetValue(fieldId, out column);
            case SortTypeIndicator:
                return columnMappings.Indicators.TryGetValue(fieldId, out column);
            case SortTypeLocation:
                return columnMappings.LocationLevels.TryGetValue(fieldId, out column);
            default:
                return false;
        }
```

as TryGetValue will be returning the appropriate `true` or `false` that you need in each of these cases.

Reply:
Nice indeed. I've gone further and made it a switch expression as well:

```cs
return fieldType switch
{
    SortTypeFilter => columnMappings.Filters.TryGetValue(fieldId, out column),
    SortTypeIndicator => columnMappings.Indicators.TryGetValue(fieldId, out column),
    SortTypeLocation => columnMappings.LocationLevels.TryGetValue(fieldId, out column),
    _ => false
};
```

Reply:
:drooling_face: Nice!